types/twilio-video/index.d.ts

```diff
@@ -103,9 +104,9 @@ export class LocalParticipant extends Participant {
     tracks: Map<Track.SID, LocalTrackPublication>;
     videoTracks: Map<Track.SID, LocalVideoTrackPublication>;
 
-    publishTrack(track: LocalTrack): Promise<LocalTrackPublication>;
     publishTrack(
-        mediaStreamTrack: MediaStreamTrack, options?: LocalTrackOptions,
+        track: LocalTrack | MediaStreamTrack,
```

Comment:
We need to keep the overload and pass different options as the second argument is of different types.

1. If the first arg is `LocalTrack` it will be [`LocalTrackPublishOptions`](https://media.twiliocdn.com/sdk/js/video/releases/2.3.0/docs/global.html#LocalTrackPublishOptions)
2. If the first arg is `MediaStreamTrack` it will be [`MediaStreamTrackPublishOptions`](https://media.twiliocdn.com/sdk/js/video/releases/2.3.0/docs/global.html#MediaStreamTrackPublishOptions)

https://media.twiliocdn.com/sdk/js/video/releases/2.3.0/docs/LocalParticipant.html#publishTrack

Note that `MediaStreamTrackPublishOptions` extends `LocalTrack` while `LocalTrackPublishOptions` doesn't.